std/algorithm/mutation.d

```diff
@@ -1071,6 +1071,14 @@ Params:
         copy is performed.
 */
 void move(T)(ref T source, ref T target)
+if (__traits(compiles, target = T.init))
+{
+    moveImpl(target, source);
+}
+
+deprecated("Can't move into `target` as `T` can't be assigned")
+void move(T)(ref T source, ref T target)
+if (!__traits(compiles, target = T.init))
```

Comment:
```d
template move(T) 
{
    static if (__traits(compiles, target = T.init))
    void move(ref T source, ref T target) {...}
    else
    deprecated("Can't move into `target` as `~T.stringof~` can't be assigned")
    void move(ref T source, ref T target) {...} 
}
```

Reply:
@thewilsonator Can't use IFTI when there's a `static if`. Also I think it's best not to group the two under a template block.

>     deprecated("Can't move into `target` as `~T.stringof~` can't be assigned")

Now implemented.